pkgs/by-name/pi/piped-backend/package.nix

```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  gradle,
+  makeWrapper,
+  jre,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "piped-backend";
+  version = "0-unstable-2024-07-10";
+
+  src = fetchFromGitHub {
+    owner = "TeamPiped";
+    repo = "Piped-Backend";
+    rev = "95a16effe503ae6f6ca11738c5ebfadbd703e0ac";
+    hash = "sha256-RX+XQZnAfGb5zDstPLjnVvfu6Wuk0H76VvKl1hN1zhU=";
+  };
+
+  nativeBuildInputs = [ gradle makeWrapper ];
+
+  gradleBuildTask = "shadowJar";
+
+  installPhase = ''
+    runHook preInstall
+
+    install build/libs/piped-1.0-all.jar -Dt $out/share/java/
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${jre}/bin/java "$out/bin/${meta.mainProgram}" \
+      --add-flags "-jar $out/share/java/piped-1.0-all.jar"
+  '';
+
+  mitmCache = gradle.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
+  };
+  passthru.updateDeps = gradle.updateDeps { inherit pname; };
+  passthru.updateScript = nix-update-script { extraArgs = [ "--version=branch=master" ]; };
```

Comment:
nix-update-script isn't enough here, as it doesn't update deps.json. Here's a working example for a different package:

```nix
passthru.updateScript = _experimental-update-script-combinators.sequence [
  (gitUpdater { rev-prefix = "v"; } // { attrPath = pname; })
  finalAttrs.passthru.updateDeps
];
```